sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -4432,12 +4432,14 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
       .map(_.multipartIdentifier).map(typedVisit[Seq[String]]).toSeq
     val columnsProperties = ctx.columns.multipartIdentifierProperty.asScala
       .map(x => (Option(x.options).map(visitPropertyKeyValues).getOrElse(Map.empty))).toSeq
-    val options = Option(ctx.options).map(visitPropertyKeyValues).getOrElse(Map.empty)
+    var options = Option(ctx.options).map(visitPropertyKeyValues).getOrElse(Map.empty)
+    if (indexType.nonEmpty) {
+      options = options + ("indexType" -> indexType)
+    }
 
     CreateIndex(
       createUnresolvedTable(ctx.multipartIdentifier(), "CREATE INDEX"),
       indexName,
-      indexType,
```

Comment:
Let's keep the plan unchanged. We should put `indexType` in the properties in `CreateIndexExec`.

Also let's not hardcode the property name. We can add a static constant in `SupportsIndex`, similar to `TableCatalog.PROP_PROVIDER`